tests/core/_orchestrator/test_orchestrator__submit.py

```diff
@@ -0,0 +1,492 @@
+    assert job_2.stacktrace == []
+    assert len(job_2._subscribers) == 2  # submission._update_submission_status and orchestrator._on_status_change
+    assert job_2._subscribers[0].__code__ == Submission._update_submission_status.__code__
+    assert job_2._subscribers[1].__code__ == _Orchestrator._on_status_change.__code__
+    # t2_bis or t2
+    job_2bis = jobs[2]
+    assert job_2bis.task == scenario.t_2bis or job_2bis.task == scenario.t_2
+    assert job_2bis.is_blocked()
+    assert job_2bis.submit_entity_id == scenario.id
+    assert not job_2bis.force
+    assert job_2bis.creation_date == s_time
+    assert len(job_2bis._subscribers) == 2  # submission._update_submission_status and orchestrator._on_status_change
+    assert job_2bis._subscribers[0].__code__ == Submission._update_submission_status.__code__
+    assert job_2bis._subscribers[1].__code__ == _Orchestrator._on_status_change.__code__
+    assert job_2bis.stacktrace == []
+    # t3
+    job_3 = jobs[3]
+    assert job_3.task == scenario.t_3
+    assert not job_3.force
+    assert job_3.is_blocked()
+    assert job_3.submit_entity_id == scenario.id
+    assert job_3.creation_date == s_time
+    assert job_3.stacktrace == []
+    assert len(job_3._subscribers) == 2  # submission._update_submission_status and orchestrator._on_status_change
+    assert job_3._subscribers[0].__code__ == Submission._update_submission_status.__code__
+    assert job_3._subscribers[1].__code__ == _Orchestrator._on_status_change.__code__
+
+    # Same submit_id
+    assert job_1.submit_id == job_2.submit_id == job_2bis.submit_id == job_3.submit_id
+
+    # submission is created and correct
+    assert len(_SubmissionManagerFactory._build_manager()._get_all()) == 1
+    submission = _SubmissionManagerFactory._build_manager()._get(job_1.submit_id)
+    assert submission.submission_status == SubmissionStatus.BLOCKED
+    assert submission.jobs == jobs
+    assert submission.creation_date == s_time
+    assert submission.entity_id == scenario.id
+    assert submission.entity_type == "SCENARIO"
+    assert submission.entity_config_id == "scenario_cfg"
+
+    # orchestrator state is correct
+    assert len(orchestrator.blocked_jobs) == 4
+    assert orchestrator.jobs_to_run.qsize() == 0
+
+
+def test_submit_scenario_standalone_mode():
+    Config.configure_job_executions(mode=JobConfig._STANDALONE_MODE)
+    sc = create_scenario()
+    orchestrator = _OrchestratorFactory._build_orchestrator()
+    sc.dn_0.write(0)  # input data is made ready
+    submit_time = datetime.now()
+    with freezegun.freeze_time(submit_time):
+        jobs = orchestrator.submit(sc)  # No dispatcher running. sc is not executed.
+
+    # task output should be locked for edition
+    assert sc.dn_1.edit_in_progress
+    assert sc.dn_2.edit_in_progress
+    assert sc.dn_3.edit_in_progress
+
+    # jobs are created in a specific order and are correct
+    assert len(jobs) == 4
+    # t1
+    job_1 = jobs[0]
```

Comment:
💬 I wonder if the following code could be factorized to avoid the repetitions... 
That would make the test slightly more readable IMHO.

Reply:
Yes, probably. Let me think about it.